app/models/katello/product.rb

```diff
@@ -205,9 +205,14 @@ def to_action_input
     end
 
     def cdn_resource
-      certs = { :ssl_client_cert => OpenSSL::X509::Certificate.new(certificate),
-                :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
-      ::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
+      product_certificate = certificate
+      if product_certificate.nil?
+        nil
+      else
+        certs = { :ssl_client_cert => OpenSSL::X509::Certificate.new(product_certificate),
+                  :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
+        ::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
+      end
```

Comment:
This might be a little cleaner:

```
return unless (product_certificate = certificate)
certs = { :ssl_client_cert => OpenSSL::X509::Certificate.new(product_certificate),
                :ssl_client_key => OpenSSL::PKey::RSA.new(key) }
::Katello::Resources::CDN::CdnResource.new(provider.repository_url, certs)
```